### ISSUE TITLE
box.savepoint() sometimes causes an incorrect error message

### ISSUE THREAD

**Issue by pgulutzan** [tarantool/tarantool#2748]
I'm using the server as a console.
This means that I don't have to send all requests
on one line or put them a function,
I can do individual requests.
But box.savepoint() doesn't know about this.

Example:
```
box.schema.space.create('Q')
box.space.Q:create_index('Q')
box.begin()
box.space.Q:insert{1}
s = box.savepoint()
box.rollback()
box.space.Q:select()
```
When I do this, I see:
```
tarantool> box.begin()
---
...

tarantool> box.space.Q:insert{1}
---
- [1]
...

tarantool> s = box.savepoint()
---
- error: Can not set a savepoint in absence of active transaction
...

tarantool> box.rollback()
---
...

tarantool> box.space.Q:select()
---
- []
...
```

I conclude that: box.savepoint() thinks there
is no active transaction, but box.rollback()
thinks there is, and that is correct, because
(as box.space.Q:select() proves)
the transaction is rolled back.

**Comment by Gerold103** [comment]
"I'm using the server as a console.
This means that I don't have to send all requests
on one line or put them a function,
I can do individual requests."

It is incorrect assumption. Interactive transactions do not work via console connection. You can check it, if call box.space.Q:select() right after insert():
unix/:/tmp/tarantool-test-console.sock> box.begin()
---
...

unix/:/tmp/tarantool-test-console.sock> box.space.Q:insert{1}
---
- [1]
...

unix/:/tmp/tarantool-test-console.sock> box.space.Q:select{}
---
- []
...

**Comment by pgulutzan** [comment]
Gerold103is of course correct so I apologize for my careless introductory remark. However, look at the title of this issue "box.savepoint() sometimes causes an incorrect error message" and the example. When there is no active transaction, box.savepoint() returns an error. When there is no active transaction, box.rollback() does not return an error. Why is behaviour different?

**Comment by Gerold103** [comment]
Possibly because complete rollback can not fail by definition? And because box.savepoint() returns a savepoint object which can be used to rollback to it. What must box.savepoint() return if there is no an active transaction? If it returns not real savepoint object, then box.rollback_to_savepoint() fails. If it returns nil (same as return nothing), and we allow to give nil to box.rollback_to_savepoint() and make this operation as noop, then box.rollback_to_savepoint with no arguments would work without errors. It is not clear too.
I think, it is temporary non-consistent effect, which will be fixed by #2016.

**Comment by pgulutzan** [comment]
I won't argue further about it, please close the issue in whatever way you think is appropriate.
